doc/sources/oneapi_gpu.rst

```diff
@@ -0,0 +1,107 @@
+This is done via integration with
+`dpctl <https://intelpython.github.io/dpctl/latest/index.html>`_ package that
+implements core oneAPI concepts like queues and devices.
+
+Prerequisites
+-------------
+
+For execution on GPU, DPC++ compiler runtime and driver are required. Refer to `DPC++ system
+requirements <https://software.intel.com/content/www/us/en/develop/articles/intel-oneapi-dpcpp-system-requirements.html>`_ for details.
+
+DPC++ compiler runtime can be installed either from PyPI or Anaconda:
+
+- Install from PyPI::
+
+     pip install dpcpp-cpp-rt
+
+- Install from Anaconda::
+
+     conda install dpcpp_cpp_rt -c intel
+
+Device offloading
+-----------------
+
+Intel(R) Extension for Scikit-learn* offers two options for running an algorithm on a
+specific device with the help of dpctl:
+
+1. Pass input data as `dpctl.tensor.usm_ndarray <https://intelpython.github.io/dpctl/latest/docfiles/dpctl.tensor_api.html#dpctl.tensor.usm_ndarray>`_ to the algorithm.
+
+   The computation will run on the device where the input data is
+   located, and the result will be returned as :code:`usm_ndarray` to the same
+   device.
+
+   .. note::
+     All the input data for an algorithm must reside on the same device.
+
+   .. warning::
+     The :code:`usm_ndarray` can only be consumed by the base methods
+     like :code:`fit`, :code:`predict`, and :code:`transform`.
+     Note that only the algorithms in Intel(R) Extension for Scikit-learn* support
+     :code:`usm_ndarray`. The algorithms from the stock version of scikit-learn
+     do not support this feature.
+2. Use global configurations of Intel(R) Extension for Scikit-learn\*:
+     1. The :code:`target_offload` option can be used to set the device primarily
+        used to perform computations. Accepted data types are :code:`str` and
+        :code:`dpctl.SyclQueue`. If string, expected to be "auto" (the execution
+        context is deduced from input data location), or SYCL* filter selector
+        string. Default value is "auto"
+     2. The :code:`allow_fallback_to_host` option
+        is a Boolean flag. If set to :code:`True`, the computation is allowed 
+        to fallback to the host device when a particular estimator does not support
+        the selected device. The default value is :code:`False`.
+
+These options can be set using :code:`sklearnex.set_config()` function or
+:code:`sklearnex.config_context`. To obtain the current values of these options,
+call :code:`sklearnex.get_config()`.
+
+.. note::
+     Functions :code:`set_config`, :code:`get_config` and :code:`config_context`
+     are always patched after the :code:`sklearnex.patch_sklearn()` call.
+
+**Compatibility considerations**::
+
+For compatibility reasons, algorithms in Intel(R) Extension for
+Scikit-learn* can be offloaded to the device with use of
+:code:`daal4py.oneapi.sycl_context`. It is recommended to use the ways
+described above for device offloading instead of using :code:`sycl_context`.
```

Comment:
```suggestion
:code:`daal4py.oneapi.sycl_context`. However, it is recommended to use one of the options
described above for device offloading instead of using :code:`sycl_context`.
```